ui/widget.js

```diff
@@ -390,6 +409,64 @@ $.Widget.prototype = {
 		return this._setOptions({ disabled: true });
 	},
 
+	_classes: function( options ) {
+		var full = [],
+			that = this,
+			settings = $.extend( {}, {
+				element: this.element,
+				keys: null,
+				extra: null,
+				add: false,
```

Comment:
`keys`, `extra`, and `add` are all unnecessary here. The defaults are all falsey and only used if truthy.